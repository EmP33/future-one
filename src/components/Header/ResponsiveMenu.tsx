import React from "react";
import { Wrapper } from "./ResponsiveMenu.styles";
import { Link } from "react-router-dom";
import { IoMdClose } from "react-icons/io";

interface IProps {
  showMenu: boolean;
  toggleShowMenu: () => void;
}

const ResponsiveMenu: React.FC<IProps> = ({ showMenu, toggleShowMenu }) => {
  return (
    <Wrapper showMenu={showMenu}>
      <button onClick={toggleShowMenu}>
        <IoMdClose role="button" />
      </button>
      <ul>
        <li>
          <Link to="/" className="active">
            Overview
          </Link>
        </li>
        <li>
          <Link to="/">About</Link>
        </li>
        <li>
          <Link to="/">Resources</Link>
        </li>
        <li>
          <Link to="/">Network</Link>
        </li>
        <li>
          <Link to="/">Cloud</Link>
        </li>
        <li>
          <Link to="/">Download</Link>
        </li>
        <li>
          <Link to="/">Contact</Link>
        </li>
      </ul>
    </Wrapper>
  );
};

export default ResponsiveMenu;
