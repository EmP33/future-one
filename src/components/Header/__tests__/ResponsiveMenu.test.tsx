import ResponsiveMenu from "../ResponsiveMenu";
import ReactShallowRenderer from "react-test-renderer/shallow";
import renderer from "react-test-renderer";
import styled from "styled-components";
import "jest-styled-components";

const StyledResponsiveMenu = styled.nav<{ showMenu: boolean }>`
  width: 100vw;
  height: 100vh;
  position: fixed;
  background-color: var(--color-green);
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  transition: all 0.2s linear;
  transform: ${({ showMenu }) =>
    showMenu ? "translateX(0)" : "translateX(100%)"};

  @media only screen and (min-width: 1200px) {
    display: none;
  }
`;

describe("<ResponsiveMenu />", () => {
  it("should render properly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(
      <ResponsiveMenu showMenu={true} toggleShowMenu={jest.fn()} />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("should have transform translateX set on 0 when showMenu is true", () => {
    const menu = renderer
      .create(<StyledResponsiveMenu showMenu={true} />)
      .toJSON();
    expect(menu).toHaveStyleRule("transform", "translateX(0)");
  });

  it("should have transform translateX set on 100% when showMenu is false", () => {
    const menu = renderer
      .create(<StyledResponsiveMenu showMenu={false} />)
      .toJSON();
    expect(menu).toHaveStyleRule("transform", "translateX(100%)");
  });
});
