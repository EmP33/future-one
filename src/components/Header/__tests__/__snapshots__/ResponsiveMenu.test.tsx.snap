// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponsiveMenu /> should render properly 1`] = `
<styled.nav
  showMenu={true}
>
  <button
    onClick={[MockFunction]}
  >
    <IoMdClose
      role="button"
    />
  </button>
  <ul>
    <li>
      <Link
        className="active"
        to="/"
      >
        Overview
      </Link>
    </li>
    <li>
      <Link
        to="/"
      >
        About
      </Link>
    </li>
    <li>
      <Link
        to="/"
      >
        Resources
      </Link>
    </li>
    <li>
      <Link
        to="/"
      >
        Network
      </Link>
    </li>
    <li>
      <Link
        to="/"
      >
        Cloud
      </Link>
    </li>
    <li>
      <Link
        to="/"
      >
        Download
      </Link>
    </li>
    <li>
      <Link
        to="/"
      >
        Contact
      </Link>
    </li>
  </ul>
</styled.nav>
`;
